File: src/game/catalogue/catalogue_page.c
#include <stdlib.h>
#include <string.h>

#include "communication/messages/outgoing_message.h"
#include "game/player/player.h"

#include "catalogue_page.h"

/**
 *
 * @param id
 * @param min_role
 * @param name_index
 * @param name
 * @param layout
 * @param image_headline
 * @param image_teasers
 * @param body
 * @param label_pick
 * @param label_extra_s
 * @param label_extra_t
 * @return
 */
catalogue_page *catalogue_page_create(int id, int min_role, char *name_index, char *name, char *layout, char *image_headline, char *image_teasers, char *body, char *label_pick, char *label_extra_s, char *label_extra_t) {
    catalogue_page *page = malloc(sizeof(catalogue_page));
    page->id = id;
    page->min_role = min_role;
    page->name_index = name_index;
    page->name = name;
    page->layout = layout;
    page->image_headline = image_headline;
    page->image_teasers = image_teasers;
    page->body = body;
    page->label_pick = label_pick;
    page->label_extra_s = label_extra_s;
    page->label_extra_t = label_extra_t;
    return page;
}

File: Kepler-Server/src/main/java/org/alexdev/kepler/game/catalogue/CataloguePage.java
package org.alexdev.kepler.game.catalogue;

public class CataloguePage {
    private int id;
    private int minRole;
    private String nameIndex;
    private String name;
    private String layout;
    private String imageHeadline;
    private String imageTeasers;
    private String body;
    private String labelPick;
    private String labelExtra_s;
    private String labelExtra_t;

    public CataloguePage(int id, int minRole, String nameIndex, String name, String layout, String imageHeadline, String imageTeasers, String body, String labelPick, String labelExtra_s, String labelExtra_t) {
        this.id = id;
        this.minRole = minRole;
        this.nameIndex = nameIndex;
        this.name = name;
        this.layout = layout;
        this.imageHeadline = imageHeadline;
        this.imageTeasers = imageTeasers;
        this.body = body;
        this.labelPick = labelPick;
        this.labelExtra_s = labelExtra_s;
        this.labelExtra_t = labelExtra_t;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getMinRole() {
        return minRole;
    }

    public void setMinRole(int minRole) {
        this.minRole = minRole;
    }

    public String getNameIndex() {
        return nameIndex;
    }

    public void setNameIndex(String nameIndex) {
        this.nameIndex = nameIndex;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getLayout() {
        return layout;
    }

    public void setLayout(String layout) {
        this.layout = layout;
    }

    public String getImageHeadline() {
        return imageHeadline;
    }

    public void setImageHeadline(String imageHeadline) {
        this.imageHeadline = imageHeadline;
    }

    public String getImageTeasers() {
        return imageTeasers;
    }

    public void setImageTeasers(String imageTeasers) {
        this.imageTeasers = imageTeasers;
    }

    public String getBody() {
        return body;
    }

    public void setBody(String body) {
        this.body = body;
    }

    public String getLabelPick() {
        return labelPick;
    }

    public void setLabelPick(String labelPick) {
        this.labelPick = labelPick;
    }

    public String getLabelExtra_s() {
        return labelExtra_s;
    }

    public void setLabelExtra_s(String labelExtra_s) {
        this.labelExtra_s = labelExtra_s;
    }

    public String getLabelExtra_t() {
        return labelExtra_t;
    }

    public void setLabelExtra_t(String labelExtra_t) {
        this.labelExtra_t = labelExtra_t;
    }
}
